Sources/AWSSignerV4/signer.swift

```diff
@@ -141,28 +142,28 @@ public struct AWSSigner {
             self.signedHeaders = signedHeadersArray.sorted().joined(separator: ";")
         }
     }
-    
+
     // Stage 3 Calculating signature as in https://docs.aws.amazon.com/general/latest/gr/sigv4-calculate-signature.html
     func signature(signingData: SigningData) -> String {
-        let kDate = hmac(string:signingData.date, key:Array("AWS4\(credentials.secretAccessKey)".utf8))
-        let kRegion = hmac(string: region, key: kDate)
-        let kService = hmac(string: name, key: kRegion)
-        let kSigning = hmac(string: "aws4_request", key: kService)
-        let kSignature = hmac(string: stringToSign(signingData: signingData), key: kSigning)
-        return AWSSigner.hexEncoded(kSignature)
+        let kDate = HMAC<SHA256>.authenticationCode(for: Data(signingData.date.utf8), using: SymmetricKey(data: Array("AWS4\(credentials.secretAccessKey)".utf8)))
+        let kRegion = HMAC<SHA256>.authenticationCode(for: Data(region.utf8), using: SymmetricKey(data: kDate))
+        let kService = HMAC<SHA256>.authenticationCode(for: Data(name.utf8), using: SymmetricKey(data: kRegion))
+        let kSigning = HMAC<SHA256>.authenticationCode(for: Data("aws4_request".utf8), using: SymmetricKey(data: kService))
+        let kSignature = HMAC<SHA256>.authenticationCode(for: stringToSign(signingData: signingData), using: SymmetricKey(data: kSigning))
+        return kSignature.hexDigest()
     }
-    
+
     /// Stage 2 Create the string to sign as in https://docs.aws.amazon.com/general/latest/gr/sigv4-create-string-to-sign.html
-    func stringToSign(signingData: SigningData) -> String {
+    func stringToSign(signingData: SigningData) -> Data {
         let stringToSign = "AWS4-HMAC-SHA256\n" +
             "\(signingData.datetime)\n" +
             "\(signingData.date)/\(region)/\(name)/aws4_request\n" +
-            AWSSigner.hexEncoded(sha256(canonicalRequest(signingData: signingData)))
-        return stringToSign
+            SHA256.hash(data: canonicalRequest(signingData: signingData)).hexDigest()
+        return Data(stringToSign.utf8)
```

Comment:
We don't need to use Data here, do we? `stringToSign.utf8` should be an [UInt8] anyway.